modules/ding_place2book/lib/p2b/P2b.php

```diff
@@ -559,7 +559,7 @@ private function p2bRequest($url, $type, array $params = array(), $code = 200) {
     );
 
     if (!empty($params)) {
-      if ($type == 'POST') {
+      if ($type == 'POST' || $type == 'PUT') {
```

Comment:
Seems unrelated?

Reply:
As mentioned in the documentation http://developer.place2book.com/v1/events/#update-an-event. Header options should be of the type json

Reply:
Sorry, I just can see how this solves the problem of default settings overriding actual values?

Reply:
when the node form is saved for the first time the POST request is sent to the p2b server with the `passive` property in a json object. This property is received on GET request and assigend in the line 180(https://github.com/ding2/ding2/pull/1472/files#diff-8f3bc34f993926671354976be58fae7aR180). 
But when the node is updated the PUT method is used. If no Content-Type json is set then it seems that an empty object is sent. So, the entity is not updated on the p2b side.

Reply:
Ok, thanks for elaborating. So the problem wasn't really defaults overriding, but that the event was never updated?

When looking at how we determine the passive state now it just looks weird:

```
      $passive = $event->passive ?? 0;
      if (!isset($meta_data->event_state)) {
        $passive = variable_get('ding_p2b_default_passive', FALSE);
      }
      elseif ($meta_data->event_state == 'passive') {
        $passive = 1;
      }
```

So it looks like passive state can be found two places and that the one on$meta_data->event_state takes precedence. And if the there's nothing set on $meta_data->event_state it always fallback to default value regardless of the value on @event->passive.

Is this really correct? I would think one of them would be the correct one to get the passive state.

Reply:
It seems that the problem is not in the module but in the place2book service. [See video](https://www.dropbox.com/s/utreqbku41wg87h/p2b.mp4?dl=0). When event is firstly created it saves all the needed fields. But when a price is created for a specific event then it recives the published status.

Reply:
One more thing, when `$event->passive == TRUE` then `$meta_data->event_state == 'passive'` and `$event->state == 'passive'`.

Reply:
> One more thing, when `$event->passive == TRUE` then `$meta_data->event_state == 'passive'` and `$event->state == 'passive'`.

Ok. Why not just check one of them then? It seems excessive and needlessly complex to check them all.

Reply:
> It seems that the problem is not in the module but in the place2book service. [See video](https://www.dropbox.com/s/utreqbku41wg87h/p2b.mp4?dl=0). When event is firstly created it saves all the needed fields. But when a price is created for a specific event then it receives the published status. Is this on purpose?

Thanks for the vid. Really helpful 👍 

I guess it's because it doesn't make sense to have tickets on a passive event, so p2b automatically publishes the event. So yeah, I think it's on purpose.

Reply:
If so, then the fix for the node update should be enough, I guess :) Everything else was reverted to initial state.